aws/resource_aws_config_aggregator_test.go

```diff
@@ -0,0 +1,257 @@
+					resource.TestCheckResourceAttr("aws_config_aggregator.example", "organization_aggregation_source.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckConfigAggregatorName(n, desired string, obj *configservice.ConfigurationAggregator) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.Attributes["name"] != *obj.ConfigurationAggregatorName {
+			return fmt.Errorf("Expected name: %q, given: %q", desired, *obj.ConfigurationAggregatorName)
+		}
+		return nil
+	}
+}
+
+func testAccCheckConfigAggregatorExists(n string, obj *configservice.ConfigurationAggregator) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not Found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No config aggregator ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).configconn
+		out, err := conn.DescribeConfigurationAggregators(&configservice.DescribeConfigurationAggregatorsInput{
+			ConfigurationAggregatorNames: []*string{aws.String(rs.Primary.Attributes["name"])},
+		})
+		if err != nil {
+			return fmt.Errorf("Failed to describe config aggregator: %s", err)
+		}
+		if len(out.ConfigurationAggregators) < 1 {
+			return fmt.Errorf("No config aggregator found when describing %q", rs.Primary.Attributes["name"])
+		}
+
+		ca := out.ConfigurationAggregators[0]
+		*obj = *ca
+
+		return nil
+	}
+}
+
+func testAccCheckConfigAggregatorDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).configconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_config_aggregator" {
+			continue
+		}
+
+		resp, err := conn.DescribeConfigurationAggregators(&configservice.DescribeConfigurationAggregatorsInput{
+			ConfigurationAggregatorNames: []*string{aws.String(rs.Primary.Attributes["name"])},
+		})
+
+		if err == nil {
+			if len(resp.ConfigurationAggregators) != 0 &&
+				*resp.ConfigurationAggregators[0].ConfigurationAggregatorName == rs.Primary.Attributes["name"] {
+				return fmt.Errorf("config aggregator still exists: %s", rs.Primary.Attributes["name"])
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccConfigAggregatorConfig_account(rString string) string {
+	return fmt.Sprintf(`
+resource "aws_config_aggregator" "example" {
+  name = "tf-%s"
+
+  account_aggregation_source {
+    account_ids = ["${data.aws_caller_identity.current.account_id}"]
+    regions     = ["us-west-2"]
+  }
+}
+
+data "aws_caller_identity" "current" {}
+`, rString)
+}
+
+func testAccConfigAggregatorConfig_organization(rString string) string {
+	return fmt.Sprintf(`
+resource "aws_config_aggregator" "example" {
+  depends_on = ["aws_iam_role_policy_attachment.example"]
+
+  name = "tf-%s"
+
+  organization_aggregation_source {
```

Comment:
_MAINTAINERS NOTE:_ This will currently create an AWS Organizations organization, enable service control policy on it, and leave it dangling after the test. Will need to add the Terraform `aws_organizations_organization` resource to this test configuration and serialize with the rest of the AWS Organizations testing.